template/golang-middleware/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM golang:1.10.4-alpine3.8
```

Comment:
Go 1.11 has an issue with DNS as I remember?

Shall we update both Go versions in this repo to use the same version?

Reply:
In Go 1.11 you can't do the build inside alpine since the http package switched to modules and requires gcc. If you change the image to 1.11 you'll see that it crashes with `exec: "gcc": executable file not found in $PATH`